.travis.yml

```diff
@@ -0,0 +1,69 @@
+#
+# Turn off email notifications entirely
+#
+notifications:
+  email: false
+
+language: generic
+
+#
+# Define the build matrix
+#
+matrix:
+  include:
+#    - services: docker
+#      env: OS_RELEASE=debian_unstable
+#      sudo: required
+#      script:
+#        - docker-compose -f scripts/docker/docker-compose.yml build $OS_RELEASE
+#        - docker-compose -f scripts/docker/docker-compose.yml run $OS_RELEASE /bin/bash /workspace/code/scripts/travis.sh
+    - os: osx
+      osx_image: xcode7.3
+    - os: osx
+      osx_image: xcode7
+    - os: osx
+      osx_image: beta-xcode6.1
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      brew update &&
+      brew tap caskroom/cask &&
+      brew outdated cmake || brew install cmake &&
+      brew outdated swig || brew install swig &&
+      brew outdated gobject-introspection || brew install gobject-introspection &&
+      brew outdated python3 || brew install python3 &&
+      brew outdated python || brew install python &&
+      brew outdated pygobject3 || brew install pygobject3 --with-python --with-python3 &&
+      brew outdated lua || brew install lua &&
+      brew outdated yajl || brew install yajl &&
+      brew outdated augeas || brew install augeas &&
+      brew install Caskroom/cask/java &&
+      brew outdated libffi || brew install libffi &&
+      libffi_ver=$(brew ls --versions libffi | awk '{print $2}') && PKG_CONFIG_PATH=/usr/local/Cellar/libffi/${libffi_ver}/lib/pkgconfig/ /usr/local/bin/luarocks install lgi &&
+      export JAVA_HOME=$(/usr/libexec/java_home -v 1.8) &&
+      java -version &&
+      echo $JAVA_HOME;
+      export PATH=/usr/local/bin:/usr/local/sbin:$PATH;
+    fi
+#
+# Source is checked out in $TRAVIS_BUILD_DIR
+# Build directory will be TRAVIS_BUILD_DIR/../build
+#
+before_script:
+  - cd $TRAVIS_BUILD_DIR/..
+  - mkdir build && cd build
+  - CMAKE_OPT=()
+  - echo $PATH
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      python2_ver=$(python2 -c 'import sys; print(".".join(map(str, sys.version_info[:2])))') &&
+      CMAKE_OPT+=("-DPYTHON2_INCLUDE_DIR:PATH=$(python2-config --prefix)/include/python${python2_ver}") &&
+      CMAKE_OPT+=("-DPYTHON2_LIBRARY:FILEPATH=$(python2-config --prefix)/lib/libpython${python2_ver}.dylib") &&
+      python3_ver=$(python3 -c 'import sys; print(".".join(map(str, sys.version_info[:2])))') &&
+      CMAKE_OPT+=("-DPYTHON_INCLUDE_DIR:PATH=$(python3-config --prefix)/include/python${python3_ver}m") &&
+      CMAKE_OPT+=("-DPYTHON_LIBRARY:FILEPATH=$(python3-config --prefix)/lib/libpython${python3_ver}.dylib");
```

Comment:
While this does fix the builds on travis I would suggest/like to solve the same with cmake. It would make building on OS X a litte less painful.

The nice thing with travis is: we now have a raw description of the build dependencies which we could document in `doc/COMPILE.md` or somewhere.